services/src/messaging/src/main/java/org/openkilda/messaging/model/SwitchId.java

```diff
@@ -0,0 +1,132 @@
+/* Copyright 2017 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.messaging.model;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Represents a switch id entity.
+ */
+public class SwitchId implements Serializable, Comparable<SwitchId> {
+
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * Switch switchId.
+     */
+    private final long switchId;
+
+    /**
+     * Instance constructor.
+     *
+     * @param switchId switch id.
+     */
+    public SwitchId(String switchId) {
+
+        String[] bytesString = switchId.split("[-:]");
+
+        long value = 0;
+        for (String oneByte : bytesString) {
```

Comment:
Why `Long.parseLong(String s, int radix)` is not suitable here?

Reply:
This will not work, because Long is a sign type and 8 bytes can not fit in Long this way because of the sign bit.

Reply:
If you try do this:
        String string = "de:ad:be:ef:00:00:00:0a";
        long number = Long.parseLong(string, 16);
        System.out.println(number);
You will get this error:
Exception in thread "main" java.lang.NumberFormatException: For input string: "de:ad:be:ef:00:00:00:0a"

Reply:
If you try do this:

        String string = "deadbeef0000000a";
        long number = Long.parseLong(string, 16);
        System.out.println(number);

You'll get this error:
Exception in thread "main" java.lang.NumberFormatException: For input string: "deadbeef0000000a"

But if you try to take a number that will be less everything will be ok:

        String string = "1eadbeef0000000a";
        long number = Long.parseLong(string, 16);
        System.out.println(number);

Printed value: 2210632925815242762

Reply:
Therefore, to put the 8-byte number in long, we must use shifts.